packages/streamid/src/event-id.ts

```diff
@@ -0,0 +1,202 @@
+}
+
+export class EventID {
+  protected readonly _tag = TAG
+
+  private readonly _networkID: number
+  private readonly _separator: Uint8Array
+  private readonly _controller: Uint8Array
+  private readonly _init: Uint8Array
+  private readonly _eventHeight: number
+  private readonly _event: CID
+
+  // WORKARDOUND Weird replacement for Symbol.hasInstance due to
+  // this old bug in Babel https://github.com/babel/babel/issues/4452
+  // which is used by CRA, which is widely popular.
+  static isInstance(instance: any): instance is EventID {
+    return typeof instance === 'object' && '_tag' in instance && instance._tag === TAG
+  }
+
+  /**
+   * Create a new EventID.
+   */
+  constructor(networkID: number, separator: Uint8Array, controller: Uint8Array, init: Uint8Array, eventHeight: number, event: CID) {
+    this._networkID = networkID
+    this._separator = separator
+    this._controller = controller
+    this._init = init
+    this._eventHeight = eventHeight
+    this._event = event
+  }
+
+  /**
+   * Create a new EventID.
+   */
+  static create(networkID: number | string, separator: string, controller: string, init: CID | string, eventHeight: number, event: CID | string ): EventID {
+    const networkIDInt = typeof networkID === 'string' ? networkByName(networkID) : networkID
+    const separatorBytes = sha256(u8a.fromString(separator)).slice(-8)
+    const controllerBytes = sha256(u8a.fromString(controller)).slice(-8)
+    const initCid = typeof init === 'string' ? CID.parse(init) : init
+    const initBytes = initCid.bytes.slice(-4)
+    const eventCid = typeof event === 'string' ? CID.parse(event) : event
+    return new EventID(networkIDInt, separatorBytes, controllerBytes, initBytes, eventHeight, eventCid)
+  }
+
+  /**
+   * Bytes representation of EventID.
+   */
+  @Memoize()
+  get bytes(): Uint8Array {
+    const streamCodec = varint.encode(STREAMID_CODEC)
+    const dagcborCodec = varint.encode(DAG_CBOR_CODEC)
+    const networkID = varint.encode(this._networkID)
+    const eventHeight = cbor.encode(this._eventHeight)
+    const event = this._event.bytes
+    return u8a.concat([streamCodec, dagcborCodec, networkID, this._separator, this._controller, this._init, eventHeight, event])
+  }
+
+  /**
+   * EventID instance from bytes
+   */
+  static fromBytes(bytes: Uint8Array): EventID {
+    try {
+      const netOffset = STREAMID_CODEC_LEN + DAG_CBOR_CODEC_LEN
+      const networkID = varint.decode(bytes, netOffset)
+      const sepOffset = varint.decode.bytes + netOffset
+      const seperator = bytes.slice(sepOffset, sepOffset + 8)
+      const conOffset = sepOffset + 8 
+      const controller = bytes.slice(conOffset, conOffset + 8)
+      const initOffest = conOffset + 8 
+      const init = bytes.slice(initOffest, initOffest + 4)
+      const ehOffset = initOffest + 4 
+      const remaining = bytes.slice(ehOffset)
+      const eventHeight = cbor.decode(bytes.slice(ehOffset, ehOffset + remaining.length - 36))
```

Comment:
I guess cbor decode does not tell you how many bytes it consumed but this seems a little fragile. I am also not sure if we need this function. The only question for an EventID is "What is the CID?" so we can look it up in IPFS. After that I don't know if we need full parse.

Reply:
yeah agree, not great for decoding (var length), but as mentioned seemed like going from bytes to parts is not that important, other than grabbing cid off, which assume is always going to be 36 bytes in ceramic 

did we use cbor instead of varint to save a byte?

Reply:
We used cbor instead of varint here so that the sorting bytes is the same as sorting heights. Varint don't sort in the order that the actual ints do,

Reply:
ah ok, makes sense